pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -1151,9 +1151,34 @@ func validateParamRef(pr *admissionregistration.ParamRef, fldPath *field.Path) f
 	if pr == nil {
 		return allErrors
 	}
-	for _, msg := range path.ValidatePathSegmentName(pr.Name, false) {
-		allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), pr.Name, msg))
+
+	if len(pr.Name) > 0 {
+		for _, msg := range path.ValidatePathSegmentName(pr.Name, false) {
+			allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), pr.Name, msg))
+		}
+
+		if pr.Selector != nil {
+			allErrors = append(allErrors, field.Forbidden(fldPath.Child("name"), `name and selector are mutual exclusive`))
+		}
 	}
+
+	if pr.Selector != nil {
+		labelSelectorValidationOpts := metav1validation.LabelSelectorValidationOptions{}
+		allErrors = append(allErrors, metav1validation.ValidateLabelSelector(pr.Selector, labelSelectorValidationOpts, fldPath.Child("selector"))...)
+
+		if len(pr.Name) > 0 {
+			allErrors = append(allErrors, field.Forbidden(fldPath.Child("selector"), `name and selector are mutual exclusive`))
+		}
+	}
+
+	if len(pr.ParameterNotFoundAction) > 0 {
```

Comment:
Check if ParameterNotFoundAction is set and return required err if not. Add test accordingly as well.

Reply:
The current implementation returns a `field.Required` error if `len(ParameterNotFoundAction)` == 0. Test is here:

https://github.com/kubernetes/kubernetes/blob/2a809d9011b330304fee9b6a380a3a7ea11d20b8/pkg/apis/admissionregistration/validation/validation_test.go#L3906-L3921